### PR TITLE
MAUI-5269 : Prepare UG for Working with MVVM and Modal window in .NET MAUI popup

### PR DESCRIPTION
**Pull request by JayashreeRavishankar**
Prepare UG for Working with MVVM and Modal window in .NET MAUI popup

### CODE REVIEW

File: MAUI/Popup/modal-window.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+title: Modal Window in .NET MAUI Popup control | Syncfusion
+description: Learn here all about Modal Window support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
+platform: MAUI
+control: SfPopup
+documentation: ug
+---
+
+# Modal Window in MAUI Popup (SfPopup)
+
+You can use popup as modal window by using the built-in Close icon and the `SfPopup.StaysOpen` property prevents interaction with your application until you close the window.
```
Comment (jayaleshwari) [inline MAUI/Popup/modal-window.md:12]: Need cpas here for Close

File: MAUI/Popup/modal-window.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+{% tabs %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+private void ClickToShowPopup_Clicked(object sender, EventArgs e)
+{
+    sfPopup.IsOpen = true;
+}
+{% endhighlight %}
+{% endtabs %}
+
+![.NET MAUI Popup as Modal Window](Images/modal-window/maui-popup-modal-window.png)
```
Comment (jayaleshwari) [inline MAUI/Popup/modal-window.md:97]: change alt text to .NET MAUI Popup as modal window

File: MAUI/Popup/working-with-mvvm.md [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+title: MVVM in .NET MAUI Popup control | Syncfusion
+description: Learn here all about MVVM support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
+platform: MAUI
+control: SfPopup
+documentation: ug
+---
+
+# Working with MVVM in MAUI Popup (SfPopup)
+
+`SfPopup` can be used in MVVM architecture applications easily. In the below example `SfPopup.IsOpen` property is bound to a property in the ViewModel based on which the popup is opened or closed. Refer the below code example to display popup in MVVM.
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:12]: highlight ViewModel

File: MAUI/Popup/working-with-mvvm.md [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+
+In the below code snippet, note that the Binding context is set for the page and the property (DisplayPopup) of the ViewModel is bound to the `SfPopup.IsOpen`.
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:14]: use BindingContext as tech word. or use binding context

File: MAUI/Popup/working-with-mvvm.md [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    <ContentPage.BindingContext>
+        <local:ViewModel/>
+    </ContentPage.BindingContext>
+    <ContentPage.Content>
+        <StackLayout Padding="20">
+            <Button x:Name="clickToShowPopup" Text="ClickToShowPopup" 
+                    VerticalOptions="Start" HorizontalOptions="Center" 
+                    Command="{Binding OpenPopupCommand}" />
+            <sfPopup:SfPopup x:Name="sfPopup"
+                             IsOpen="{Binding DisplayPopup}">
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:32]: use property name as like bool property

File: MAUI/Popup/working-with-mvvm.md [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+            RaisePropertyChanged("DisplayPopup");
+        }
+    }
+
+    public ViewModel()
+    {
+        OpenPopupCommand = new Command(OpenPopup);
+    }
+
+    private void OpenPopup()
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:65]: OnOpenPopup

File: MAUI/Popup/working-with-mvvm.md [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+    public string Password
+    {
+        get { return password; }
+        set
+        {
+            password = value;
+            OnPropertyChanged(nameof(Password));
+        }
+    }
+    public bool OpenPopup
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:228]: use bool convention for property name

File: MAUI/Popup/working-with-mvvm.md [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+
+    #endregion
+
+    public ViewModel()
+    {
+        LoginCommand = new Command(Login);
+    }
+
+    // Executed when the login button is clicked.
+    private void Login()
```
Comment (jayaleshwari) [inline MAUI/Popup/working-with-mvvm.md:246]: OnLogin

File: MAUI/Popup/modal-window.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: Modal Window in .NET MAUI Popup control | Syncfusion
+description: Learn here all about Modal Window support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
```
Comment (florenceliliana) [inline MAUI/Popup/modal-window.md:4]: description: Learn here all about Modal Window support in the Syncfusion .NET MAUI Popup (SfPopup) control, its elements, and more.

File: MAUI/Popup/modal-window.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+title: Modal Window in .NET MAUI Popup control | Syncfusion
+description: Learn here all about Modal Window support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
+platform: MAUI
+control: SfPopup
+documentation: ug
+---
+
+# Modal Window in MAUI Popup (SfPopup)
+
+You can use popup as modal window by using the built-in close icon and the [SfPopup.StaysOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_StaysOpen) property prevents interaction with your application until you close the window.
```
Comment (florenceliliana) [inline MAUI/Popup/modal-window.md:12]: You can use the popup as a modal window using the built-in close icon and the [SfPopup.StaysOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_StaysOpen) property prevents interaction with your application until you close the window.

File: MAUI/Popup/modal-window.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+`Modal`: Window loads under the parent window surrounded by an overlay which prevents clicking anywhere else on the screen apart from the control of the modal.
+
+Modal does not require any action to open. It opens in the same window and gives callbacks when closing or opening the window.
+
+Refer to the following code example in which the popup will close only if you click on close icon.
```
Comment (florenceliliana) [inline MAUI/Popup/modal-window.md:18]: Refer to the following code example: the popup will only close if you click on the close icon.

File: MAUI/Popup/modal-window.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+        <sfPopup:SfPopup x:Name="sfPopup"     
+                         StaysOpen="True"
+                         HeaderTitle="Modal Window"
+                         ShowCloseButton="True"
+                         WidthRequest="312"
+                         HeightRequest="180"
+                         HeaderHeight="72">
+            <sfPopup:SfPopup.ContentTemplate>
+                <DataTemplate>
+                    <Label Text="A modal window disables the parent window while the user interacts with the child (modal) window before they return to the parent application."
```
Comment (florenceliliana) [inline MAUI/Popup/modal-window.md:40]: <Label Text= "A modal window disables the parent window while the user interacts with the child (modal) window before they return to the parent application."

File: MAUI/Popup/working-with-mvvm.md [hunk 13]
```diff
@@ -0,0 +1,444 @@
+---
+layout: post
+title: MVVM in .NET MAUI Popup control | Syncfusion
+description: Learn here all about MVVM support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:4]: description: Learn here all about MVVM support in the Syncfusion .NET MAUI Popup (SfPopup) control, its elements, and more.

File: MAUI/Popup/working-with-mvvm.md [hunk 14]
```diff
@@ -0,0 +1,444 @@
...
+title: MVVM in .NET MAUI Popup control | Syncfusion
+description: Learn here all about MVVM support in Syncfusion .NET MAUI Popup (SfPopup) control, its elements and more.
+platform: MAUI
+control: SfPopup
+documentation: ug
+---
+
+# Working with MVVM in MAUI Popup (SfPopup)
+
+`SfPopup` can be used in MVVM architecture applications easily. In the below example [SfPopup.IsOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_IsOpen) property is bound to a property in the `ViewModel` based on which the popup is opened or closed. Refer the below code example to display popup in MVVM.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:12]: `SfPopup` can be used in the MVVM architecture applications easily. In the below example, the [SfPopup.IsOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_IsOpen) property is bound to a property in the `ViewModel` based on which the popup is opened or closed. Refer to the following code example to display the popup in MVVM.

File: MAUI/Popup/working-with-mvvm.md [hunk 15]
```diff
@@ -0,0 +1,444 @@
...
+
+In the below code snippet, note that the BindingContext is set for the page and the property (IsOpen) of the ViewModel is bound to the [SfPopup.IsOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_IsOpen).
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:14]: In the following code sample, note that the BindingContext is set for the page, and the property (IsOpen) of the ViewModel is bound to the [SfPopup.IsOpen](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Popup.SfPopup.html#Syncfusion_Maui_Popup_SfPopup_IsOpen).

File: MAUI/Popup/working-with-mvvm.md [hunk 16]
```diff
@@ -0,0 +1,444 @@
...
+
+`SfPopup` provides command support for the popup buttons,
+
+ * [SfPopup.AcceptCommand](https://help.syncfusion.com/maui/popup/popup-events#accept-command)
+
+ * [SfPopup.DeclineCommand](https://help.syncfusion.com/maui/popup/popup-events#decline-command)
+
+## How to use MAUI popup as a page?
+
+To use `SfPopup` as a page, create a new XAML file with `SfPopup` in it. You can use this XAML file instance (SfPopup view), across any file in your project wherever you may want to display a popup, thus allowing you to reuse the same instance of the popup where you can modify or change the required properties alone based on the current context to be displayed.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:97]: To use the `SfPopup` as a page, create a new XAML file with `SfPopup`. You can use this XAML file instance (SfPopup view) across any file in your project wherever you may want to display a popup, thus allowing you to re-use the same instance of the popup where you can modify or change the required properties alone based on the current context to be displayed.

File: MAUI/Popup/working-with-mvvm.md [hunk 17]
```diff
@@ -0,0 +1,444 @@
...
+
+The below code snippet shows how to create a popup in XAML file. We have created a simple popup to show error message popup to user on a unsuccessful login.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:99]: The following code sample shows how to create a popup in a XAML file. We have created a simple popup to display an error message popup to users on an unsuccessful login.

File: MAUI/Popup/working-with-mvvm.md [hunk 18]
```diff
@@ -0,0 +1,444 @@
...
+{
+    public PopupPage()
+    {
+        InitializeComponent();
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+The below code snippet shows how to consume the previously created SfPopup page from inside a different content page.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:137]: The following code sample shows how to consume the previously created SfPopup page from inside a different content page.

File: MAUI/Popup/working-with-mvvm.md [hunk 19]
```diff
@@ -0,0 +1,444 @@
...
+    #endregion
+}
+{% endhighlight %}
+{% endtabs %}
+
+## How to retrieve text value from entry control in MAUI popup content template?
+
+This can be done by binding the text property of the entry to a property in the view model class.
+
+Refer to the following code example to retrieve text value from entry control in MAUI popup content template.
```
Comment (florenceliliana) [inline MAUI/Popup/working-with-mvvm.md:274]: Refer to the following code example to retrieve text value from entry control in the MAUI popup content template.

**Review by florenceliliana — CHANGES_REQUESTED** [review]
The PR review is completed.

**Review by jayaleshwari — APPROVED** [review]
This is fine

**Pull request merged**